test/factories/delegatedManagerFactory.spec.ts

```diff
@@ -0,0 +1,707 @@
+  getRandomAddress,
+} from "@utils/index";
+
+import { ProtocolUtils } from "@utils/common";
+import { SystemFixture } from "@setprotocol/set-protocol-v2/utils/fixtures";
+import { MODULE_STATE } from "@setprotocol/set-protocol-v2/utils/constants";
+
+import {
+  getSystemFixture,
+  getProtocolUtils
+} from "@setprotocol/set-protocol-v2/utils/test";
+import { SetToken } from "@setprotocol/set-protocol-v2/utils/contracts";
+
+const expect = getWaffleExpect();
+
+describe("DelegatedManagerFactory", () => {
+  let owner: Account;
+  let methodologist: Account;
+  let otherAccount: Account;
+  let operatorOne: Account;
+  let operatorTwo: Account;
+  let EOAManagedSetToken: SetToken;
+
+  let setV2Setup: SystemFixture;
+
+  let deployer: DeployHelper;
+  let protocolUtils: ProtocolUtils;
+
+  let delegatedManagerFactory: DelegatedManagerFactory;
+  let mockFeeExtension: BaseGlobalExtensionMock;
+  let mockIssuanceExtension: BaseGlobalExtensionMock;
+
+  cacheBeforeEach(async () => {
+    [
+      owner,
+      otherAccount,
+      methodologist,
+      operatorOne,
+      operatorTwo,
+    ] = await getAccounts();
+
+    deployer = new DeployHelper(owner.wallet);
+    protocolUtils = getProtocolUtils();
+
+    setV2Setup = getSystemFixture(owner.address);
+    await setV2Setup.initialize();
+
+    mockFeeExtension = await deployer.mocks.deployBaseGlobalExtensionMock();
+    mockIssuanceExtension = await deployer.mocks.deployBaseGlobalExtensionMock();
+
+    delegatedManagerFactory = await deployer.factories.deployDelegatedManagerFactory(
+      setV2Setup.factory.address
+    );
+  });
+
+  // Helper function to run a setup execution of either `createSetAndManager` or `createManager`
+  async function create(module: Address, extension: Address, existingSetToken?: Address): Promise<ContractTransaction> {
+    const tokens = [setV2Setup.dai.address, setV2Setup.wbtc.address];
+    const operators = [operatorOne.address, operatorTwo.address];
+    const otherAccountAddress = otherAccount.address;
+    const methodologistAddress = methodologist.address;
+
+    if (existingSetToken === undefined) {
+      return await delegatedManagerFactory.createSetAndManager(
+        tokens,
+        [ether(1), ether(.1)],
+        "TestToken",
+        "TT",
+        otherAccountAddress,
+        methodologistAddress,
+        [module],
+        operators,
+        tokens,
+        [extension]
+      );
+    }
+
+    return await delegatedManagerFactory.createManager(
+      existingSetToken as string,
+      otherAccountAddress,
+      methodologistAddress,
+      operators,
+      tokens,
+      [extension]
+    );
+  }
+
+  // Helper function to generate bytecode packets for factory initialization call
+  async function generateBytecode(setToken: Address, manager: Address): Promise<string[]> {
+    const iFace = new ethersUtils.Interface([
+      "function initialize(address,address)",
+      "function initializeExtension(address,address)"
+    ]);
```

Comment:
Should really use `basicIssuanceModule.interface.encodeFunctionData("initialize" ...` for this.

Reply:
[5086ffe](https://github.com/SetProtocol/set-v2-strategies/pull/14/commits/5086ffeb824bb3b117c26c5c414b0fb4068ae37f)